### PR TITLE
Implimented until valid Input and testing for io

### PR DESCRIPTION
**Pull request by hkgumbs**

### CODE REVIEW

File: Sources/Specs/EndGameDetectorSpec.swift [hunk 1]
```diff
@@ -35,6 +35,10 @@ class EndGameDetectorSpec: Group {
         let board = Board();
         assert.that(!EndGameDetector.isWin(board: board))
     }
+    let noTieTest = Test { assert in
+        let board = Board();
+        assert.that(!EndGameDetector.isTie(board: board))
```
Comment (hkgumbs) [inline Sources/Specs/EndGameDetectorSpec.swift]: 😄

File: Sources/Specs/HumanSpec.swift [hunk 2]
```diff
@@ -2,8 +2,20 @@ import XTest
...
     let getMoveReturnsAnIntTest = Test { assert in
-        let human = Human(marker:"X")
+        let testio = testIO(inputs: ["lol","1"])
+        let io: IO = testio
+        let human = Human(marker:"X", io: io)
         assert.that(human.getMove(avaliableMoves: [1]), equals: 1)
+        assert.that(testio.outputs[0], equals: "Select a Move:")
+        assert.that(testio.outputs[1], equals: "Invalid Input Please Try Again")
+        assert.that(testio.outputs[2], equals: "Select a Move:")
+        assert.that(testio.outputs.count, equals: 3)
```
Comment (hkgumbs) [inline Sources/Specs/HumanSpec.swift]: Beatuiful! I think this is a much clearer test than the nested Test-Mock thing we had going on earlier. Simplest thing that could possibly work!

**Comment by hkgumbs** [comment]
I mentioned that I would comment on here with a pattern to help you avoid this kind of function: `func validate(prompt: String) -> (Bool, Int?)`. It's called an `Either` or a `Maybe` or a `Result` (depending on language or framework), and it's super easy to implement:
```swift
enum Result {
  case Unsuccessful
  case Successful(Int)
}


// now your function signiature can look like this:
func validate(prompt: String) -> Result


// and you use it by switching on the enum (yay type safety)
let result = validate(prompt: "Enter a move")
switch result {
case .Unsuccessful:
  // handle error case
case .Successful(let value):
  // do something with `value`
}
```

File: Sources/Specs/HumanSpec.swift (hunk 2 above)
Comment (Jgoodrich07) [inline Sources/Specs/HumanSpec.swift]: +1 nice job!

**Pull request merged**